index.html

```diff
@@ -1321,12 +1294,85 @@
       <p>
         The following example shows how to represent US$55.00.
       </p>
-      <pre class="example">
-{
-  "currency": "USD",
-  "value" : "55.00"
-}
+      <pre class="example js">
+        {
+          "currency": "USD",
+          "value" : "55.00"
+        }
       </pre>
+      <section>
+        <h3>
+          Validity checkers
+        </h3>
+        <p>
+          A <dfn data-cite="INFRA#javascript-string">JavaScript string</dfn> is
+          a <dfn>valid decimal monetary value</dfn> if it consists of the
+          following <dfn data-lt="code point" data-cite="INFRA#code-point">code
+          points</dfn> in the given order: [[!INFRA]]
+        </p>
+        <ol>
+          <li>Optionally, a single U+002D HYPHEN-MINUS (-), to indicate that
+          the amount is negative.
+          </li>
+          <li>One or more <a>code points</a> in the range U+0030 DIGIT ZERO (0)
+          to U+0039 DIGIT NINE (9).
+          </li>
+          <li>Optionally, a single U+002E FULL STOP (.) followed by one or more
+          <a>code points</a> in the range U+0030 DIGIT ZERO (0) to U+0039 DIGIT
+          NINE (9).
+          </li>
+        </ol>
+        <div class="note">
+          The following regular expression is an implementation of the above
+          definition.
+          <pre class="hljs">^-?[0-9]+(\.[0-9]+)?$</pre>
+        </div>
+        <p>
+          To <dfn data-lt="check and canonicalize amount">check and
+          canonicalize <var>amount</var> <a>PaymentCurrencyAmount</a></dfn>,
+          run the following steps:
+        </p>
+        <ol data-link-for="PaymentCurrencyAmount">
+          <li>If <a>currencySystem</a> is not
```

Comment:
It should be _amount_.currencySystem. Same for `currency` and "_amount_'s `value`" below.